### ISSUE TITLE
OTP PIN blacklist

### ISSUE THREAD

**Issue by cornelinux** [privacyidea/privacyidea#1428]
We already have otp pin contents policies:
https://privacyidea.readthedocs.io/en/latest/policies/user.html#otp-pin-contents

Why not provide an OTP PIN blacklist?

The blacklist could contain the compromised passwords of major password leaks.
Or it could contain customer specific entries.
Administrators could manage the blacklist.

The question is:

* Plain file?
* plain entries or hashed passwords?
* how to handle the blacklisted passwords? In the token database, so that it gets synced in an HA setup?
   * in a berkley DB or redis?
* Add passwords to blacklist via command line or REST API?

**Comment by plettich** [comment]
What about checking against a different (internal) service?
How do i.e. Linux distros check for bad passwords? Maybe we can utilize these tools.

**Comment by cornelinux** [comment]
Yes, there is an online-service from haveIbeenpwned, that fetches part of the hashes of passwords.
I think it would be great if we could combine both.

**Comment by plettich** [comment]
https://www.heise.de/security/artikel/Nach-dem-Passwort-Leak-Eigene-Passwoerter-lokal-checken-4284756.html

**Comment by cornelinux** [comment]
@plettich the mentioned lists have roughly 1G of data. We do not want to ship this.

I am thinking about dropping this. But on the other hand there are several project were we actually also do some "user management". So why not also improve password/PIN management....
